File: Framework/Framework.Klowd/Client/fw/features/component.js
// ============================================================================
// Project: Framework
// Name/Class: Component feature.
// Created On: 27/Mar/2016
// Author: João Carreiro (joao.carreiro@cybermap.pt)
// Company: Cybermap Lda.
// ============================================================================

'use strict';
fw.feature('component', {

    value: function (deps, def) {

        //
        // Component value definitions are objects
        // that contain the following properties:
        //
        //    o- base :: string
        //    o- description :: string
        //    o- properties :: object
        //    o- model :: object
        //    o- resources :: object 
        //    o- events :: object
        //    o- template :: object | string
        //    o- placeholders :: object
        //    o- native : function
        //

        var component = null;

        if (typeof def == 'object') {

            component = def;
            component.deps = deps;            
        }

        return component;
    }

});

File: Framework/Framework.Design/Model/Objects/Client/Component.cs
// ============================================================================
// Project: Framework
// Name/Class: 
// Author: João Carreiro (joao.carreiro@coop4creativity.com)
// Create date: 18/Jul/2016
// Company: Coop4Creativity
// Description: 
// ============================================================================

using Framework.Core.Types.Specialized;
using Framework.Core.Patterns;

namespace Framework.Design.Model.Objects.Client
{
    public class Block<TUser> : ABaseClassWithID<Id, TUser>
    {
        //
        // PROPERTIES
        //

        public Id Type { get; set; }

        //
        // CONSTRUCTORS
        // 

        public Block() : base()
        {
            Type = default(Id);
        }
    }
}
